### DISCUSSION TITLE
part-ii/2019-25/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#37] [Question comments]
Part II 2019 Automata and Formal Languages: Paper 2, Section I, H 

https://questions.tripos.org/part-ii/2019-25/

**Comment by callumhobbis** [comment]
<details>
<summary>Sketch solution</summary>

This is a quick one to write up for me.

**(a):** Definitions and proof given in lectures (lectures 5, 6 in 2021-22).

**(b):** Also in lectures! (lecture 6)

**(c):** This was example sheet 1, Q3(a) (as of 2021-22). It's a simple recursion with $f(0)=0$ as the base case and $f(n+1)=g(n,f(n))$ where $g(n,f(n))=n$ is the projection function. Everything's primitive so $f$ is too.

</details>
